meson.build

```diff
@@ -0,0 +1,172 @@
+project(
+  'turbodbc', 'cpp',
+  license: 'MIT',
+  meson_version: '>= 1.1.0',
+  default_options: [
+    'buildtype=release',
+    'cpp_std=c++17',
+    'warning_level=3',
+    'build_testing=false',
+  ]
+)
+
+fs = import('fs')
+
+cpp = meson.get_compiler('cpp')
+add_global_arguments('-DBOOST_NO_CXX98_FUNCTION_BASE', language: 'cpp')
+add_global_arguments('-DGTEST_LINKED_AS_SHARED_LIBRARY=1', language: 'cpp')
+if target_machine.system() == 'windows'
+  add_global_arguments('-DNOMINMAX', language: 'cpp')
+endif
+simdutf = dependency('simdutf')
+arrow = dependency('arrow')
+build_testing = get_option('build_testing')
+if build_testing
+  gtest = dependency('gtest')
+  gtest_main = dependency('gtest_main')
+  gmock = dependency('gmock')
+endif
+
+# unixodbc = dependency('unixodbc')
+py = import('python').find_installation()
+
+# Taken from https://github.com/scipy/scipy/blob/e2ce1a57639dbfa5d1439cb542b67e3cbeda5d29/scipy/meson.build#L30
+# The chdir is needed because within numpy there's an `import signal`
+# statement, and we don't want that to pick up scipy's signal module rather
+# than the stdlib module. The try-except is needed because when things are
+# split across drives on Windows, there is no relative path and an exception
+# gets raised. There may be other such cases, so add a catch-all and switch to
+# an absolute path. Relative paths are needed when for example a virtualenv is
+# placed inside the source tree; Meson rejects absolute paths to places inside
+# the source tree.
+# For cross-compilation it is often not possible to run the Python interpreter
+# in order to retrieve numpy's include directory. It can be specified in the
+# cross file instead:
+#   [properties]
+#   numpy-include-dir = /abspath/to/host-pythons/site-packages/numpy/core/include
+#
+# This uses the path as is, and avoids running the interpreter.
+incdir_numpy = meson.get_external_property('numpy-include-dir', 'not-given')
+if incdir_numpy == 'not-given'
+  incdir_numpy = run_command(py,
+    [
+      '-c',
+      '''import os
+import numpy as np
+try:
+  incdir = os.path.relpath(np.get_include())
+except Exception:
+  incdir = np.get_include()
+print(incdir)
+  '''
+    ],
+    check: true
+  ).stdout().strip()
+
+  # We do need an absolute path to feed to `cc.find_library` below
+  _incdir_numpy_abs = run_command(py,
+    ['-c', 'import numpy; print(numpy.get_include())'],
+    check: true
+  ).stdout().strip()
+else
+  _incdir_numpy_abs = incdir_numpy
+endif
+inc_np = include_directories(incdir_numpy)
+# Don't use the deprecated NumPy C API. Define this to a fixed version instead of
+# NPY_API_VERSION in order not to break compilation for released SciPy versions
+# when NumPy introduces a new deprecation.
+numpy_nodepr_api = ['-DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION']
```

Comment:
I see this is copied from SciPy, but I wonder why NPY_1_9_API_VERSION (or NPY_1_7_API_VERSION) and not the minimum numpy?